processing/src/main/java/org/apache/druid/data/input/InputSource.java

```diff
@@ -87,4 +88,9 @@ InputSourceReader reader(
       @Nullable InputFormat inputFormat,
       File temporaryDirectory
   );
+
+  @Nullable
+  default Set<String> getTypes() {
```

Comment:
Maybe should add this function to a different interface, with non-default, and have the tasks that are now implementing properly, implement that interface, so its easier to know which tasks are properly returning this information